### PR TITLE
WildCam: discard Carto, use Datasette

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview

Requires: https://github.com/zooniverse/labs-datasette-api/ (new API)
Context: the Carto map database requirement (for WildCam Labs) is being retired. The map database will be replaced by a Zooniverse-hosted Datasette solution.

This PR updates WildCam Gorongosa Lab and WildCam Darien Lab to integrate with the new Labs Datasette API.

Goals:
- [x] An explorer should be able to explore the map for **both Darien and Gorongosa**, e.g. http://localhost:3998/#/wildcam-darien-lab/educators/map
  - [x] An explorer should be able to see "circles" (cameras) representing results
  - [x] An explorer should be able to see "shapes" of park boundaries, etc
  - [x] An explorer should be able to use filters (Species, Habitats, etc) to narrow down results
  - [x] An explorer should be able to view the number of results (at the top of the Filter controls)
  - [x] An explorer should be able to click on a circle (camera) to see the camera's related data.
  - [x] An explorer should be able to download the results as a CSV
- [x] A teacher should be able to select a number of Subjects when creating Assignments.
  - Corollary: A teacher should be able to create Assignments. i.e. make sure to test submitting the Create Assignment action to the API!

### Status

WIP

### CODE REVIEW

**Comment by shaunanoordin** [comment]
⚠️ This PR's feature complete, BUT...

- This update shouldn't be deployed until we have the Datasette API deployed to production. The database URLs for `wildcam-XYZ.map-config.js` are currently pointing to localhost.
- There's a hidden limit on the database API that limits the number of rows that can be returned. (See the [settings.json](https://github.com/zooniverse/labs-datasette-api/blob/main/settings.json) file.) Since this limit is rather large (2000 at time of writing), it will ONLY have practical effects on the "Download as CSV" and "Select for Assignment" actions. (Even then, I'm not sad if an assignment can "only" have 2000 subjects. It's not practical for students to classify that many!)
  - We can either crank this limit up to a huge number on the API on deployment, or **spell out the limit** on the front end. I'm thinking that spelling out the limit is partially unavoidable.

File: webpack.config.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = {
   devServer: {
     contentBase: path.join(__dirname, '/src/'),
     disableHostCheck: true,  //Allow access using custom host names, e.g. local.zooniverse.org:3998, which is necessary for testing certain API calls.
-    https: true,
+    // https: true,
```
Comment (camallen) [inline webpack.config.js:15]: Not blocking: Should leave this as https by default and change it locally if we need to? 
FWIW i had to disable https for local http API integration dev / testing. Once we deploy the labs-datasette-api it'll be on https so this setting won't impact.

**Review by camallen — COMMENTED** [review]
This LGTM - i'll get the labs datasette ready for deployment so we can get real world URLs for this PR to use vs localhost

**Comment by camallen** [comment]
Classroom maps API deployment PR https://github.com/zooniverse/classroom-maps-api/pull/3
once that is out the service will be deployed at https://classroom-maps-api.zooniverse.org

**Comment by camallen** [comment]
@shaunanoordin datasette labs API is deployed at https://classroom-maps-api.zooniverse.org/ 🎉 

We can update this PR and test again that service directly via HTTPS too :)

File: webpack.config.js (hunk 1 above)
Comment (shaunanoordin) [reply webpack.config.js]: Yup, I've reverted this back to `https: true`

**Comment by shaunanoordin** [comment]
## PR Update

Alright, this PR is now ready to roll! WildCam Darien Lab and WildCam Gorongosa Lab are now pulling from the live https://classroom-maps-api.zooniverse.org/ URL (emphasis on the https) and everything is looking good!

### Status

Merging, go!

Follow up:
- Need to add the 2000 maximum Subjects selection/download limit notification. This is a separate job.

**Pull request merged**